src/Workspaces/Core/MSBuild.BuildHost/BuildHost.cs

```diff
@@ -118,29 +122,29 @@ private void CreateBuildManager()
                 _logger.LogInformation($"Logging builds to {_binaryLogPath}");
             }
 
-            _buildManager = new ProjectBuildManager(ImmutableDictionary<string, string>.Empty, logger);
-            _buildManager.StartBatchBuild();
+            _buildManager = new ProjectBuildManager(_globalMSBuildProperties, logger);
+            _buildManager.StartBatchBuild(_globalMSBuildProperties);
         }
     }
 
-    public Task<bool> HasUsableMSBuildAsync(string projectOrSolutionFilePath, CancellationToken cancellationToken)
+    public bool HasUsableMSBuild(string projectOrSolutionFilePath)
     {
-        return Task.FromResult(TryEnsureMSBuildLoaded(projectOrSolutionFilePath));
+        return TryEnsureMSBuildLoaded(projectOrSolutionFilePath);
     }
 
     private void EnsureMSBuildLoaded(string projectFilePath)
     {
-        Contract.ThrowIfFalse(TryEnsureMSBuildLoaded(projectFilePath), $"We don't have an MSBuild to use; {nameof(HasUsableMSBuildAsync)} should have been called first to check.");
+        Contract.ThrowIfFalse(TryEnsureMSBuildLoaded(projectFilePath), $"We don't have an MSBuild to use; {nameof(HasUsableMSBuild)} should have been called first to check.");
     }
 
-    public Task<ImmutableArray<(string ProjectPath, string ProjectGuid)>> GetProjectsInSolutionAsync(string solutionFilePath, CancellationToken cancellationToken)
+    public ImmutableArray<(string ProjectPath, string ProjectGuid)> GetProjectsInSolution(string solutionFilePath)
```

Comment:
The mini RPC library implemeted here is less strict about requiring things to be Task-returning, so this is just updating to the simpler pattern. (It originally started not having async support at all until I realized I needed it.)

Reply:
note: if these still are RPC calls, i would *very* much like them to be Task-based.  that way we understand *virally* where async work (like cross process calls) are happening, and we have to code appropriately.

Reply:
This change was reverted and they're task-returning again.